src/components/molecules/denomination-grid/index.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useMemo, useRef, useState } from "react"
+import { useObserveWidth } from "../../../hooks/use-observe-width"
+import DenominationBadge from "../../atoms/denomination-badge"
+
+export type GiftCardVariant = {
+  prices: {
+    currency_code: string
+    amount: number
+  }[]
+}
+
+type DenominationGridProps = {
+  variants: GiftCardVariant[]
+  defaultCurrency: string
+}
+
+const DenominationGrid: React.FC<DenominationGridProps> = ({
+  variants,
+  defaultCurrency,
+}) => {
+  const containerRef = useRef(null)
+
+  const width = useObserveWidth(containerRef)
+
+  const [columns, setColumns] = useState(7)
+
+  // We estimate that a Badge is rouglhy 70px and use this to calulate how many we can fit in a row
+  // It is a bit of a "magic" number, but it works for now.
+  useEffect(() => {
+    if (width) {
+      setColumns(Math.floor(width / 70) - 1)
+    }
+  }, [width])
+
+  const defaultDenominations = useMemo(() => {
+    return (
+      variants.map(variant => {
+        const price = variant.prices.find(
+          price => price.currency_code === defaultCurrency
+        )
+        return { amount: price?.amount, currencyCode: defaultCurrency }
+      }) ?? []
+    )
+  }, [variants, defaultCurrency])
```

Comment:
nit: what do you think about moving this up to the `GiftCardBanner` component and making the `DenominationGrid` only take a prop called `denominations` (or any better name) of type `Array<{ amount: number, currencyCode: string }>`? I feel like `DenominationGrid` shouldn't have to know about how to extract a price from a variant. The `useMemo` might also be an indicator that it might fit better somewhere else. Let me know what you think :smile:

Reply:
definitely, I have moved it to GiftCardBanner and made it return a string array that we pass along to the now renamed TagGrid